services/121-service/src/registration/validator/registration-data-type.validator.ts

```diff
@@ -0,0 +1,152 @@
+import { RegistrationEntity } from '../registration.entity';
+import {
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+  ValidationArguments,
+  ValidationOptions,
+  registerDecorator,
+} from 'class-validator';
+import { getConnection, Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import {
+  AnswerTypes,
+  CustomAttributeType,
+} from '../enum/custom-data-attributes';
+
+@ValidatorConstraint({ name: 'validateAttributeType', async: true })
+export class RegistrationDataTypeValidator
+  implements ValidatorConstraintInterface {
+  private message: string;
+
+  public constructor() {}
+
+  public async validate(
+    value: any,
+    args: ValidationArguments,
+  ): Promise<boolean> {
+    const referenceId = args.object[args.constraints[0]['referenceId']];
+    const attribute = args.object[args.constraints[0]['attribute']];
+
+    if (referenceId === undefined || attribute === undefined) {
+      this.message = 'ReferenceId or attribute are undefined';
+      return false;
+    }
+    const registrationRepository = getConnection().getRepository(
+      RegistrationEntity,
+    );
+    const registration = await registrationRepository.findOne({
+      where: { referenceId: referenceId },
+      relations: ['program'],
+    });
+    if (!registration) {
+      this.message = `Registration was not found for referenceId: '${referenceId}'`;
+      return false;
+    }
+    const validationInfo = await registration.program.getValidationInfoForQuestionName(
+      attribute,
+    );
+    if (!validationInfo.type) {
+      this.message = `Attribute '${attribute}' was not found for the program related to the reference id:' ${referenceId}'`;
+      return false;
+    }
+    return this.typeIsValid(
+      value,
+      validationInfo.type,
+      validationInfo.options,
+      attribute,
+    );
+  }
+
+  private createErrorMessageType(type, value, attribute): void {
+    const valueString = Array.isArray(value) ? JSON.stringify(value) : value;
+    this.message = `The value '${valueString} given for the attribute '${attribute}' does not have the corect type: '${type}'`;
+  }
+
+  public defaultMessage(args: ValidationArguments): string {
+    return this.message;
+  }
+
+  private typeIsValid(
+    value: any,
+    type: string,
+    options: any[],
+    attribute: string,
+  ): boolean {
+    let isValid = false;
+    if (type === AnswerTypes.date) {
+      const parsedDate = Date.parse(value);
```

Comment:
@RubenGeo don't parse as date, but treat as string that needs to follow specified pattern